### PR TITLE
Refactored bucket.gd

### PR DESCRIPTION
**Pull request by MatteoPiovanelli-Laser**
Separated out the larger functions to ease readability and possible future improvements.

### CODE REVIEW

File: src/Tools/Bucket.gd [hunk 1]
```diff
@@ -348,47 +349,59 @@ func _flood_fill(position: Vector2) -> void:
...
+	for j in image.get_height():
+		_add_new_segment(j)
+	# start flood algorithm
+	_flood_line_around_point(position, project, image, src_color)
+	# test all segments while also discovering more
+	var done = false
+	while not done:
+		done = true
+		var max_index = _allegro_flood_segments.size()
+		#print(max_index)
```
Comment (OverloadedOrama) [inline src/Tools/Bucket.gd:371]: You forgot a commented print here :)

Comment (MatteoPiovanelli-Laser) [reply src/Tools/Bucket.gd:371]: Who? Me? Impossible!
Now, late-night-commit-me, you shouldn't trust that guy's code...

(thanks for spotting this)

**Comment by MatteoPiovanelli-Laser** [comment]
Just for context, the point of the refactor in this PR is to do this:
image
and be able to find what needs improving (those are function times on a bucket fill on a 3000x3000 px image)

**Review by OverloadedOrama — APPROVED** [review]
Yup, that sounds very helpful indeed. Thanks a lot!

**Pull request merged**